<a href="https://colab.research.google.com/github/lucksza/datascience-cp2/blob/main/datascience_cp2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 1. Introdução e contextualização da base de dados
o Descrição da base e do problema analisado

o Objetivo da análise

In [ ]:
import pandas as pd

pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', 5)
pd.set_option('display.width', None)

df = pd.read_csv('rh.csv')
df


Nome     ID  É Casado (0=Não, 1=Sim)  \
0         Adinolfi, Wilson  K  10026                        0   
1    Ait Sidi, Karthikeyan     10084                        1   
..                        ...    ...                      ...   
309               Zhou, Julia  10043                        0   
310             Zima, Colleen  10271                        0   

     ID do Estado Civil  ID do Gênero (0=Feminino, 1=Masculino)  \
0                     0                                       1   
1                     1                                       1   
..                  ...                                     ...   
309                   0                                       0   
310                   4                                       0   

     ID do Status do Funcionário  ID do Departamento  \
0                              1                   5   
1                              5                   3   
..                           ...                 ...   
309                            1                   3   
310                            1                   5   

     ID da Nota de Desempenho  Veio de Feira de Diversidade (0=Não, 1=Sim)  \
0                           4                                            0   
1                           3                                            0   
..                        ...                                          ...   
309                         3                                            0   
310                         3                                            0   

     Salário (Anual)  Termd  PositionID                 Position State   Zip  \
0              62506      0          19  Production Technician I    MA  1960   
1             104437      1          27                  Sr. DBA    MA  2148   
..               ...    ...         ...                      ...   ...   ...   
309            89292      0           9             Data Analyst    MA  2148   
310            45046      0          19  Production Technician I    MA  1730   

          DOB Sex MaritalDesc CitizenDesc HispanicLatino RaceDesc DateofHire  \
0    07/10/83  M       Single  US Citizen             No    White   7/5/2011   
1    05/05/75  M      Married  US Citizen             No    White  3/30/2015   
..        ...  ..         ...         ...            ...      ...        ...   
309  02/24/79   F      Single  US Citizen             No    White  3/30/2015   
310  08/17/78   F     Widowed  US Citizen             No    Asian  9/29/2014   

    DateofTermination         TermReason        EmploymentStatus  \
0                 NaN  N/A-StillEmployed                  Active   
1           6/16/2016      career change  Voluntarily Terminated   
..                ...                ...                     ...   
309               NaN  N/A-StillEmployed                  Active   
310               NaN  N/A-StillEmployed                  Active   

            Department  Nome do Gestor  ID do Gestor Origem do Recrutamento  \
0    Production         Michael Albert          22.0               LinkedIn   
1                IT/IS      Simon Roup           4.0                 Indeed   
..                 ...             ...           ...                    ...   
309              IT/IS      Simon Roup           4.0      Employee Referral   
310  Production          David Stanley          14.0               LinkedIn   

    Desempenho (Texto)  Engajamento (Escala de 0 a 5)  \
0              Exceeds                           4.60   
1          Fully Meets                           4.96   
..                 ...                            ...   
309        Fully Meets                           5.00   
310        Fully Meets                           4.50   

     Satisfação (Escala de 1 a 5)  Projetos Especiais (Qtd.)  \
0                               5                          0   
1                               3                          6   
..                            ...           

**Descrição da Base de Dados (RH)**

**A base contém informações detalhadas de colaboradores de uma empresa, incluindo dados demográficos, contratuais e comportamentais. São 36 colunas, entre elas:**


-Identificação e demografia: Nome, Gênero, Estado Civil, Raça, Idade.

-Contratuais: Departamento, Cargo, Salário Anual, Status de Emprego, Origem do Recrutamento.

-Indicadores de desempenho e satisfação: Nota de Desempenho, Engajamento (0–5), Satisfação (1–5), Faltas, Projetos Especiais, Atrasos.

-Histórico de RH: Data de contratação, data de desligamento (se houver), motivo da demissão, gestor responsável.

--------------------------------------------------------------------------------

--------------------------------------------------------------------------------

**Objetivo da análise**

**O objetivo é identificar padrões quantitativos que indiquem fatores críticos que afetam o engajamento e a satisfação dos funcionários. A análise buscará responder perguntas como:**

-Existe uma relação entre salário e satisfação?

-Funcionários com mais projetos especiais tendem a ter menos faltas ou maior engajamento?

-Há departamentos ou tipos de contratação mais propensos a baixa performance ou alta rotatividade?

-A partir desses insights, pretende-se propor melhorias em políticas de RH focadas em retenção de talentos, aumento de produtividade e redução de absenteísmo.

# 2. Tipos de variáveis e estrutura dos dados


o Classificação das variáveis (nominais, ordinais, discretas,
contínuas)


o Compreensão do papel de cada variável no problema

In [ ]:
import pandas as pd
df = pd.read_csv('rh.csv')
for i in df.columns:
  print(i)

Nome
ID
É Casado (0=Não, 1=Sim)
ID do Estado Civil
ID do Gênero (0=Feminino, 1=Masculino)
ID do Status do Funcionário
ID do Departamento
ID da Nota de Desempenho
Veio de Feira de Diversidade (0=Não, 1=Sim)
Salário (Anual)
Termd
PositionID
Position
State
Zip
DOB
Sex
MaritalDesc
CitizenDesc
HispanicLatino
RaceDesc
DateofHire
DateofTermination
TermReason
EmploymentStatus
Department
Nome do Gestor
ID do Gestor
Origem do Recrutamento
Desempenho (Texto)
Engajamento (Escala de 0 a 5)
Satisfação (Escala de 1 a 5)
Projetos Especiais (Qtd.)
Data da Última Avaliação de Desempenho
Dias de Atraso nos Últimos 30 Dias
Faltas (Total)


**O CLASSIFICAÇÃO DAS VARIÁVEIS (NOMINAIS, ORDINAIS, DISCRETAS, CONTÍNUAS)**

---

**Nome: Nominal**

---

**ID: Nominal**

---

**É Casado (0=Não, 1=Sim): Nominal**

---

**ID do Estado Civil: Nominal**

---

**ID do Gênero (0=Feminino, 1=Masculino): Nominal**

---

**ID do Status do Funcionário: Nominal**

---

**ID do Departamento: Nominal**

---

**ID da Nota de Desempenho: Ordinal**

---

**Veio de Feira de Diversidade (0=Não, 1=Sim): Discreta**

---

**Salário (Anual): Contínua**

---

**Termd: Nominal**

---

**PositionID: Discreta**

---

**Position: Nominal**

---

**State: Nominal**

---

**Zip: Nominal**

---

**DOB: Contínua**

---

**Sex: Nominal**

---

**MaritalDesc: Nominal**

---

**CitizenDesc: Nominal**

---

**HispanicLatino: Nominal**

---

**RaceDesc: Nominal**

---

**DateofHire: Contínua**

---

**DateofTermination: Contínua**

---

**TermReason: Nominal**

---

**EmploymentStatus: Nominal**

---

**Department: Nominal**

---

**Nome do Gestor: Nominal**

---

**ID do Gestor: Nominal**

---

**Origem do Recrutamento: Nominal**

---

**Desempenho (Texto): Ordinal**

---

**Engajamento (Escala de 0 a 5): Ordinal**

---

**Satisfação (Escala de 1 a 5): Ordinal**

---

**Projetos Especiais (Qtd.): Discreta**

---

**Data da Última Avaliação de Desempenho: Contínua**

---

**Dias de Atraso nos Últimos 30 Dias: Discreta**

---

**Faltas (Total): Discreta**

---



**O COMPREENSÃO DO PAPEL DE CADA VARIÁVEL NO PROBLEMA**




 ---
Salário (Anual) | Impacto sobre Satisfação e Engajamento

---

Engajamento (0–5) | Métrica de motivação e envolvimento

---

Satisfação (1–5) | Métrica de bem-estar e contentamento

---

Projetos Especiais (Qtd.) | Verificar se maior dedicação a projetos reduz
Faltas e aumenta Engajamento

---

Faltas (Total) | Indicador de absenteísmo influenciado por níveis de engajamento e satisfação

---

Dias de Atraso nos Últimos 30 Dias | Relação entre pontualidade, Engajamento e Satisfação

---

ID da Nota de Desempenho | Influência da performance histórica em motivação e satisfação

---


Department | Comparar métricas de engajamento, satisfação e rotatividade por área

---

EmploymentStatus | Avaliar impacto do tipo de contrato (full-time, part-time, contractor) em engajamento e retenção

---

Termd (0=Ativo,1=Desligado) | Analisar rotatividade em função de engajamento e satisfação pré-desligamento

---

Origem do Recrutamento | Verificar se canal de contratação afeta retenção e engajamento

---

Data da Última Avaliação de Desempenho | Avaliar se frequência de feedback impulsiona engajamento e satisfação


# 3. Medidas descritivas


o Cálculo e análise de média, mediana, moda, variância, desvio
padrão e amplitude

In [13]:
import pandas as pd

df = pd.read_csv('rh.csv')

def calcularr(obj, coluna):
    media = round(df[coluna].mean(), 2)
    mediana = round(df[coluna].median(), 2)
    moda = int(df[coluna].mode()[0])
    variancia = round(df[coluna].var(), 2)
    desvio_padrao = round(df[coluna].std(), 2)
    amplitude = round(df[coluna].max() - df[coluna].min(), 2)

    print(f'{obj}')
    print(f'Média:         {media:,.2f}')
    print(f'Mediana:       {mediana:,.2f}')
    print(f'Moda:          {moda:,}')
    print(f'Variância:     {variancia:,.2f}')
    print(f'Desvio Padrão: {desvio_padrao:,.2f}')
    print(f'Amplitude:     {amplitude:,.2f}\n')

variaveis = [
    ('Salário (Anual)', 'Salário (Anual)'),
    ('Engajamento (0–5)', 'Engajamento (Escala de 0 a 5)'),
    ('Satisfação (1–5)', 'Satisfação (Escala de 1 a 5)'),
    ('Projetos Especiais', 'Projetos Especiais (Qtd.)'),
    ('Faltas (Total)', 'Faltas (Total)'),
    ('Dias de Atraso (30 dias)', 'Dias de Atraso nos Últimos 30 Dias'),
    ('Nota de Desempenho', 'ID da Nota de Desempenho')
]

for obj, coluna in variaveis:
    calcularr(obj, coluna)


Salário (Anual)
Média:         69,020.68
Mediana:       62,810.00
Moda:          57,815
Variância:     632,856,381.61
Desvio Padrão: 25,156.64
Amplitude:     204,954.00

Engajamento (0–5)
Média:         4.11
Mediana:       4.28
Moda:          5
Variância:     0.62
Desvio Padrão: 0.79
Amplitude:     3.88

Satisfação (1–5)
Média:         3.89
Mediana:       4.00
Moda:          3
Variância:     0.83
Desvio Padrão: 0.91
Amplitude:     4.00

Projetos Especiais
Média:         1.22
Mediana:       0.00
Moda:          0
Variância:     5.52
Desvio Padrão: 2.35
Amplitude:     8.00

Faltas (Total)
Média:         10.24
Mediana:       10.00
Moda:          4
Variância:     34.25
Desvio Padrão: 5.85
Amplitude:     19.00

Dias de Atraso (30 dias)
Média:         0.41
Mediana:       0.00
Moda:          0
Variância:     1.68
Desvio Padrão: 1.29
Amplitude:     6.00

Nota de Desempenho
Média:         2.98
Mediana:       3.00
Moda:          3
Variância:     0.34
Desvio Padrão: 0.59
Amplitude:     3.00





---

### **SALÁRIO ANUAL**  
| **Métrica**       | **Valor**        | **Descrição**                                                              
|--------------------|------------------|-----------------------------------------------------------------------------|
| **Média**         | 69.020,68       | Distribuição salarial concentrada neste valor de referência                |
| **Mediana**       | 62.810,00       | Divisão da amostra em dois grupos de igual tamanho                         |
| **Moda**          | 57.815,00       | Valor mais recorrente na série de dados                                    |
| **Variância**     | Alta            | Dispersão alta entre os valores individuais                           |
| **Desvio Padrão** | 25.156,64       | Alta variabilidade relativa (36% da média)                                 |
| **Amplitude**     | 204.954,00      | Diferença extrema entre salários mínimos e máximos                         |

---

### **ENGAJAMENTO (ESCALA 0–5)**  
| **Métrica**       | **Valor** | **Descrição**                                  
|--------------------|-----------|------------------------------------------------|
| **Média**         | 4.11      | Nível médio próximo ao máximo da escala        |
| **Moda**          | 5         | Resposta predominante no limite superior       |
| **Desvio Padrão** | 0.79      | Variação controlada entre os participantes     |

---

### **SATISFAÇÃO (ESCALA 1–5)**  
| **Métrica**       | **Valor** | **Descrição**                                  
|--------------------|-----------|------------------------------------------------|
| **Média**         | 3.89      | Tendência para avaliações positivas            |
| **Moda**          | 3         | Concentração na classificação intermediária    |
| **Desvio Padrão** | 0.91      | Heterogeneidade nas percepções de satisfação   |

---

### **PROJETOS ESPECIAIS**  
| **Métrica**       | **Valor** | **Descrição**                                  
|--------------------|-----------|------------------------------------------------|
| **Média**         | 1.22      | Participação limitada na maioria dos casos     |
| **Moda**          | 0         | Predomínio de não participantes               |
| **Desvio Padrão** | 2.35      | Alta desigualdade na distribuição              |
| **Mediana**       | 0         | 50% dos funcionários sem participação         |

---

### **FALTAS (TOTAL)**  
| **Métrica**       | **Valor** | **Descrição**                                  
|--------------------|-----------|------------------------------------------------|
| **Média**         | 10.24     | Número médio de ausências por colaborador      |
| **Moda**          | 4         | Quantidade mais frequente de faltas            |
| **Desvio Padrão** | 5.85      | Discrepância relevante entre os casos          |
| **Amplitude**     | 19        | Diferença máxima observada na amostra          |

---

### **DIAS DE ATRASO (ÚLTIMOS 30 DIAS)**  
| **Métrica**       | **Valor** | **Descrição**                                  
|--------------------|-----------|------------------------------------------------|
| **Média**         | 0.41      | Baixa incidência média de atrasos              |
| **Moda**          | 0         | Maioria sem registros de atraso                |
| **Mediana**       | 0         | Distribuição fortemente concentrada em zero    |
| **Desvio Padrão** | 1.29      | Casos pontuais com atrasos elevados            |

---

### **NOTA DE DESEMPENHO**  
| **Métrica**       | **Valor** | **Descrição**                                  
|--------------------|-----------|------------------------------------------------|
| **Média**         | 2.98      | Avaliação média próxima do centro da escala    |
| **Moda**          | 3         | Classificação mais comum entre os avaliados    |
| **Desvio Padrão** | 0.59      | Uniformidade nas avaliações de performance     |

---


# 4. Correlação


o Cálculo e visualização de correlação entre variáveis relevantes


o Interpretação contextualizada com o problema do grupo